File: WebSocketChat.Frontend/src/app/shared/models/chat/messages/nickname-request.ts
import { MessageType } from '../message-type';
import { Message, MessageIdentifier } from './../message';
export class NicknameRequest extends MessageIdentifier implements Message {

	messageType : MessageType = MessageType.NicknameRequest;

	/**The user's requested nickname */
	requestedNickname: string;

	/**Indicates whether the nickname-request was successful */
	wasSuccessful: boolean;

	/** If the user has changed his/her nickname, the old nickname will be saved here (clientside only) */
	oldNickname: string;
}
	

File: WebSocketChat.Server/WebSocketChat.Server/Models/Chat/Messages/NicknameRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebSocketChat.Server.Models.Chat.Messages
{
    public class NicknameRequest : Message
    {
        public NicknameRequest()
        {
            MessageType = MessageType.NicknameRequest;
            Timestamp = DateTime.Now;
        }

        /// <summary>
        /// Timestamp when the name was requested
        /// </summary>
        public DateTime Timestamp { get; set; }

        /// <summary>
        /// The user's requested nickname
        /// </summary>
        public string RequestedNickname { get; set; }

        /// <summary>
        /// Indicates whether the nickname-request was successful
        /// </summary>
        public bool WasSuccessful { get; set; }
    }
}
